ddtrace/context.py

```diff
@@ -52,12 +52,11 @@ def __eq__(self, other):
     def _with_span(self, span):
         # type: (Span) -> Context
         """Return a shallow copy of the context with the given span."""
-        with self._lock:
```

Comment:
Can you explain why we no longer need the lock for shallow copying the context?

Reply:
The references that are copied from the original context are written just once and then are read only. This means it's safe to copy the values without obtaining a lock since there's no risk of the references changing.